test/distributed/org/apache/cassandra/distributed/test/AuthTest.java

```diff
@@ -52,4 +64,60 @@ public void authSetupIsCalledAfterStartup() throws IOException
             assertTrue(setupCalled);
         }
     }
+
+    /**
+     * Sows that in some circumstances CassandraRoleManager will create the cassandra role twice
+     */
+    @Test
+    public void authSetupIsCalledTwice() throws IOException
+    {
+        try (Cluster cluster = builder().withNodes(1)
+                                        .withTokenSupplier(TokenSupplier.evenlyDistributedTokens(3))
+                                        .withNodeIdTopology(NetworkTopology.singleDcNetworkTopology(3, "dc0", "rack0"))
+                                        .withConfig(config -> config.with(NETWORK, GOSSIP).set("authenticator", "PasswordAuthenticator"))
+                                        .start())
+        {
+            boolean defaultRoleSetup = cluster.get(1).callOnInstance(() -> {
+                long maxWait = TimeUnit.NANOSECONDS.convert(10, TimeUnit.SECONDS);
+                long start = System.nanoTime();
+                while (!CassandraRoleManager.hasExistingRoles() && System.nanoTime() - start < maxWait)
+                    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+
+                return CassandraRoleManager.hasExistingRoles();
+            });
+            assertTrue(defaultRoleSetup);
+
+            //get the time from the the first role setup
+            Long time1 = (Long)cluster.coordinator(1).execute("SELECT WRITETIME (salted_hash) from system_auth.roles where role = 'cassandra'",
```

Comment:
formating might be better, `execute` bellow `coordinator`, `ConsistencyLevel` below `SELECT` .... you feel me